src/github.com/matrix-org/dendrite/federationsender/consumers/roomserver.go

```diff
@@ -134,25 +134,27 @@ func (s *OutputRoomEvent) processMessage(ore api.OutputNewRoomEvent) error {
 		return err
 	}
 
+	if oldJoinedHosts == nil {
```

Comment:
can we remove the TODO at  line 123?

Reply:
TBH, I don't know to what extent that TODO was:
- just for paranoia if upstream sends down a buggy message
- its intended that nodes can throw away state for a room
- there's a situation where upstream wants to send down incomplete stream of state

Basically, I'm inclined to leave it for now as I don't think this PR addresses the issue that the TODO was trying to cover

Reply:
mmkay. I'm inclined to think that a TODO we don't understand doesn't have much value; possibly it could do with clarifying.